dev: simplify --rewrite; add --stream-output

irfansharif opened the pull request:
--rewrite was previously a string flag, with the allowance to pass in
additional arguments to the test binary. We now have --test-arg for
that. While here, fix a tiny bug in dev testlogic where we ignored
cached test data when --rewrite was false, considered it when it was
true (opposite of what we wanted).

Separately, it's useful to be able to stream test output during test
runs (especially long ones).

Release justification: Non-production code
Release note: None

irfansharif committed:
dev: simplify --rewrite; add --stream-output

--rewrite was previously a string flag, with the allowance to pass in
additional arguments to the test binary. We now have --test-arg for
that. While here, fix a tiny bug in dev testlogic where we ignored
cached test data when --rewrite was false, considered it when it was
true (opposite of what we wanted).

Separately, it's useful to be able to stream test output during test
runs (especially long ones).

Release justification: Non-production code
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/78065)
<!-- Reviewable:end -->

irfansharif committed:
dev: fix testlogic selector

Gets rid of stray '/' selectors for optional fields.

Release justification: Non-production code
Release note: None

irfansharif commented:
Thanks!

bors r+
